### PR TITLE
[crunchyroll:beta] Use anonymous access instead of redirecting

### LINKED ISSUE #4692: Crunchyroll beta - KeyError('byId')
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.08.14** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

The error occurs when openening URL from the beta UI **w/o providing authentication info** to yt-dlp:
`An extractor error has occurred. (caused by KeyError('byId'));`

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://beta.crunchyroll.com/watch/G69P9MD9Y/asta-and-yuno']
[debug] Encodings: locale cp1251, fs utf-8, pref cp1251, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.08.14 [5593720] (win32_exe)
[debug] Python 3.8.10 (CPython 64bit) - Windows-10-10.0.19044-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg n4.4-19-g8d172d9409 (setts)
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.08.14, Current version: 2022.08.14
yt-dlp is up to date (2022.08.14)
[debug] [crunchyroll:beta] Extracting URL: https://beta.crunchyroll.com/watch/G69P9MD9Y/asta-and-yuno
[crunchyroll:beta] asta-and-yuno: Downloading webpage
ERROR: G69P9MD9Y: An extractor error has occurred. (caused by KeyError('byId')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "yt_dlp\extractor\common.py", line 666, in extract
  File "yt_dlp\extractor\crunchyroll.py", line 805, in _real_extract
  File "yt_dlp\extractor\crunchyroll.py", line 752, in _redirect_from_beta
KeyError: 'byId'
```

### PR DESCRIPTION
**Pull request by tejing1**
### Description of your *pull request* and other information

Eliminates the beta -> non-beta redirect code, due to changes in the api making it impossible. Instead, uses anonymous access to the beta api, which already works, and its momentarily used by the browser before redirect.

Closes #4692

Ran all beta tests, both with and without an `etp_rt` cookie.

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/crunchyroll.py [hunk 1]
```diff
@@ -720,15 +720,27 @@ class CrunchyrollBetaBaseIE(CrunchyrollBaseIE):
 
     def _get_params(self, lang):
         if not CrunchyrollBetaBaseIE.params:
+            if self._get_cookies(f'https://beta.crunchyroll.com/{lang}').get('etp_rt'):
+                auth = 'cookie'
+            else:
+                auth = 'client id'
```
Comment (pukkandan) [inline yt_dlp/extractor/crunchyroll.py:726]:
```suggestion
            if self._get_cookies(f'https://beta.crunchyroll.com/{lang}').get('etp_rt'):
                grant_type, key = 'etp_rt_cookie', 'accountAuthClientId'
            else:
                grant_type, key = 'client_id', 'anonClientId'
```
should make it  much simpler (see comment below)

File: yt_dlp/extractor/crunchyroll.py [hunk 2]
```diff
@@ -720,15 +720,27 @@ class CrunchyrollBetaBaseIE(CrunchyrollBaseIE):
...
                 f'https://beta.crunchyroll.com/{lang}', None, note='Retrieving main page'), None)
             api_domain = app_config['cxApiParams']['apiDomain']
-            basic_token = str(base64.b64encode(('%s:' % app_config['cxApiParams']['accountAuthClientId']).encode('ascii')), 'ascii')
+
+            if auth == 'client id':
+                client_id = app_config['cxApiParams']['anonClientId']
+                grant_type = 'client_id'
+            else:
+                client_id = app_config['cxApiParams']['accountAuthClientId']
+                grant_type = 'etp_rt_cookie'
```
Comment (pukkandan) [inline yt_dlp/extractor/crunchyroll.py:737]:
```suggestion
            client_id = app_config['cxApiParams'][key]
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/crunchyroll.py (hunk 1 above)
Comment (tejing1) [reply yt_dlp/extractor/crunchyroll.py:726]: `auth` is also used in the `note` for `auth_response`. I suppose I can just add it to the list of variables assigned, though.

Comment (pukkandan) [reply yt_dlp/extractor/crunchyroll.py:726]: Just use `grant_type`

Comment (tejing1) [reply yt_dlp/extractor/crunchyroll.py:726]: Hmm, ok, I suppose.

**Comment by tejing1** [comment]
Are you waiting on something from me here?

**Comment by pukkandan** [comment]
No, just been busy

**Pull request merged**